src/Security/Authorization/Core/src/AuthorizationPolicyCache.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Authorization;
+
+/// <summary>
+/// Represents a cache for an AuthorizationPolicy instance
+/// </summary>
+public class AuthorizationPolicyCache
```

Comment:
We've started using `InternalsVisibleTo` for stuff like this [example](https://github.com/dotnet/aspnetcore/pull/35408/files) that really doesn't need to be public. What do you think @davidfowl. I know you're still watching notifications. 😆 

Please make an API proposal if this does need to be public.

Reply:
Ok marking as api-suggestion for now since its public

Reply:
Where is the `api-suggestion`?

Reply:
Why would YARP need the `AuthorizationPolicyCache`? If YARP wanted it on their endpoints, couldn't the call `RequireAuthorization` on those endpoints? Or does it really want the cached `AuthorizationPolicy` instance for some reason? @Tratcher 

Do we need an API that just adds the `AuthorizationPolicyCache` to the `IEndopintConventionBuilder` and nothing else?

Reply:
That seems fine to me, so basically we just add a RequireAuthorizationPolicyCache, and call that from the existing RequireAuthZ places, so we can keep this internal?